command/info_sub/variables.sh

```diff
@@ -56,147 +55,93 @@ See also:
 
 ' 1
 
-if ! typeset -F getopts_long >/dev/null 2>&1 ; then
+if ! typeset -F getopts_long >/dev/null 2>&1; then
+    # shellcheck source=./../../getopts_long.sh
     . "${_Dbg_libdir}/getopts_long.sh"
 fi
 
-# Should declare typset_flags before calling.
-# That it the implit returned value
-function _Dbg_info_variables_parse_options {
-
-    typeset -i _Dbg_rc=0
+function _Dbg_do_info_variables {
+    declare _Dbg_typeset_flags=""
+    declare -i _Dbg_typeset_filtered=1
+    _Dbg_info_variables_parse_options "$@"
+    (($? != 0)) && return
+
+    # Create an indexed array of variable names which are matching our input flags,
+    # use a single pipeline command to avoid later evaluation of the output by Bash.
+    declare -a _Dbg_var_names=()
+    if ((_Dbg_typeset_filtered == 1)); then
+        # grepping for '=' to only accept variables with a value
+        mapfile -t _Dbg_var_names < <(declare -p $_Dbg_typeset_flags |
+            grep '=' |
```

Comment:
Previously I have been shying away from assuming POSIX command-line arguments. MS windows for example in its native mode might not have "cut" and "grep -v" and "sort". 

Creating the pipeline as a bash or zsh function should be doable. And we can test that in isolation.

Reply:
Thanks, I didn't think about Windows. I'll try to come up with equivalent Bash-only code...